src/app/providers/provider-form/github/github.component.ts

```diff
@@ -11,13 +14,24 @@ export class GithubComponent {
   providerForm: FormGroup;
 
   constructor(
+    public gitHubService: GitHubService,
     private formBuilder: FormBuilder,
     private providersService: ProvidersService) {
     this.providerForm = this.formBuilder.group(new GithubFormModel());
   }
 
   onSubmit() {
-    console.warn('Not implemented');
-    this.providersService.triggerCloseForm(this.providerForm);
+    const providerData: MainImport = {
+      url: this.providerForm.value.url,
+      token: this.providerForm.value.token,
+      group: this.providerForm.value.group,
+      project: this.providerForm.value.project,
+      uploadUrl: this.providerForm.value.uploadUrl
+    };
+    this.gitHubService.githubPost(providerData)
+      .pipe(
+        finalize(() => this.providersService.triggerCloseForm(this.providerForm))
+      )
+      .subscribe();
```

Comment:
nit: Does this need to unsubscribe when the component is destroyed? Such as if the user clicks close and then opens the model again before the request has finished

Reply:
I rewrote this part to a `pipe(first())` which will automagically handle the unsub on the response reception. Also after further consideration i moved the close logic for the providerForm to the subscribe. That way the window will not close on a sub error. See https://github.com/iver-wharf/wharf-web/pull/8/commits/5e89ca3940fd4b8f65ca52482008552d648c4963